src/main/java/de/dennisguse/opentracks/fragments/StatisticsRecordedFragment.java

```diff
@@ -24,38 +24,48 @@
 import android.view.MotionEvent;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.LinearLayout;
 import android.widget.Spinner;
 import android.widget.TextView;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProvider;
 
 import de.dennisguse.opentracks.R;
 import de.dennisguse.opentracks.TrackRecordedActivity;
-import de.dennisguse.opentracks.content.TrackDataHub;
-import de.dennisguse.opentracks.content.TrackDataListener;
 import de.dennisguse.opentracks.content.data.Track;
-import de.dennisguse.opentracks.content.data.TrackPoint;
-import de.dennisguse.opentracks.content.data.Waypoint;
+import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
 import de.dennisguse.opentracks.stats.TrackStatistics;
 import de.dennisguse.opentracks.util.PreferencesUtils;
 import de.dennisguse.opentracks.util.StringUtils;
 import de.dennisguse.opentracks.util.TrackIconUtils;
+import de.dennisguse.opentracks.viewmodels.IntervalStatisticsModel;
+import de.dennisguse.opentracks.views.IntervalListView;
 
 /**
  * A fragment to display track statistics to the user.
  *
  * @author Sandor Dornbush
  * @author Rodrigo Damazio
  */
-public class StatisticsRecordedFragment extends Fragment implements TrackDataListener {
+public class StatisticsRecordedFragment extends Fragment implements IntervalListView.IntervalListListener {
+
+    private static final String TRACK_ID_KEY = "trackId";
 
     private TrackStatistics trackStatistics = null;
     private String category = "";
+    private Track track;
+
+    private IntervalStatisticsModel viewModel = null;
```

Comment:
Why initialize some vars with `= null` directly?
Isn't implicit?

Reply:
Lack of Java experience... I won't forget anymore :smile: